tensorflow/core/kernels/mkl/onednn_nn_ops_benchmark.cc

```diff
@@ -0,0 +1,158 @@
+#include "tensorflow/core/graph/testlib.h"
+#include "tensorflow/core/public/session_options.h"
+
+namespace tensorflow {
+namespace {
+
+template <typename T>
+static void BM_oneDNN_Softmax(::testing::benchmark::State& state,
+                              std::initializer_list<int64_t> dims,
+                              int num_threads, const string& label) {
+  DataType dtype = DataTypeToEnum<T>::v();
+  TensorShape shape = TensorShape(dims);
+  Tensor input(dtype, shape);
+  input.flat<T>().setRandom();
+  Graph* g = new Graph(OpRegistry::Global());
+  test::graph::oneDNNSoftmax(g, test::graph::Constant(g, input));
+
+  SessionOptions opts;
+  opts.config.set_inter_op_parallelism_threads(1);
+  opts.config.set_intra_op_parallelism_threads(num_threads);
+  opts.config.set_use_per_session_threads(true);
+  opts.config.mutable_graph_options()
+      ->mutable_optimizer_options()
+      ->set_opt_level(OptimizerOptions::L0);
+  test::Benchmark("cpu", g, &opts, nullptr, nullptr, "",
+                  /*old_benchmark_api*/ false)
+      .Run(state);
+  state.SetItemsProcessed(shape.num_elements() * state.iterations());
+  state.SetLabel(label);
+}
+
+template <typename T>
+static void BM_Eigen_Softmax(::testing::benchmark::State& state,
+                             std::initializer_list<int64_t> dims,
+                             int num_threads, const string& label) {
+  DataType dtype = DataTypeToEnum<T>::v();
+  TensorShape shape = TensorShape(dims);
+  Tensor input(dtype, shape);
+  input.flat<T>().setRandom();
+  auto root = Scope::NewRootScope().ExitOnError();
+  auto softmax = ops::Softmax(root, input);
+  TF_CHECK_OK(root.status());
+  Graph* g = new Graph(OpRegistry::Global());
+  TF_CHECK_OK(root.ToGraph(g));
+
+  SessionOptions opts;
+  opts.config.set_inter_op_parallelism_threads(1);
+  opts.config.set_intra_op_parallelism_threads(num_threads);
+  opts.config.set_use_per_session_threads(true);
+  opts.config.mutable_graph_options()
+      ->mutable_optimizer_options()
+      ->set_opt_level(OptimizerOptions::L0);
+  test::Benchmark("cpu", g, &opts, nullptr, nullptr, "",
+                  /*old_benchmark_api*/ false)
+      .Run(state);
+  state.SetItemsProcessed(shape.num_elements() * state.iterations());
+  state.SetLabel(label);
+}
+
+/*
+ * The following trick for number of variadic macro arguments is taken from:
+ * http://groups.google.com/group/comp.std.c/browse_thread/thread/77ee8c8f92e4a3fb/346fc464319b1ee5?pli=1
+ * The macro PP_NARG returns the number of arguments that have been passed to
+ * it.  Currrently, it is prepared to support upto 5 arguments. It is straight
+ * forward to extend to a higher number of arguments.
+ */
+#define PP_NARG(...) PP_NARG_(__VA_ARGS__, PP_RSEQ_N())
+#define PP_NARG_(...) PP_ARG_N(__VA_ARGS__)
+#define PP_ARG_N(_1, _2, _3, _4, _5, N, ...) N
+#define PP_RSEQ_N() 5, 4, 3, 2, 1, 0
+
+// For a tensor shape {a, b, c, d} we want to produce a token a_b_c_d
```

Comment:
Nit: Adding a comma.
```suggestion
// For a tensor shape {a, b, c, d}, we want to produce a token a_b_c_d
```

Reply:
Done.